terraform/environments/electronic-monitoring-data/modules/landing_bucket_iam_user_access/main.tf

```
@@ -27,24 +27,64 @@ data "aws_iam_policy_document" "supplier_data_access" {
   }
 }
 
+data "aws_iam_policy_document" "rotate_iam_keys" {
+  statement {
+    sid    = "IAMKeyPermissionsForUpdatingKey${var.data_feed}${var.order_type}"
+    effect = "Allow"
+    actions = [
+      "iam:DeleteAccessKey",
+      "iam:UpdateAccessKey",
+      "iam:ListAccessKeys",
+      "iam:CreateAccessKeys"
+    ]
+    resources = [aws_iam_access_key.rotate_iam_keys.arn]
+  }
+  statement {
+    sid    = "UpdateSecretsPermissions${var.data_feed}${var.order_type}"
+    effect = "Allow"
+    actions = [
+      "secretsmanager:GetSecretValue",
+      "secretsmanager:ListSecrets",
+      "secretsmanager:UpdateSecret"
+    ]
+    resources = [module.secrets_manager.secret_arn]
+  }
+}
+
+resource "aws_iam_policy" "rotate_iam_keys" {
+  name        = "rotate-iam-keys-lambda-policy-${var.data_feed}-${var.order_type}"
+  description = "IAM policy for rotating iam keys for ${var.data_feed} ${var.order_type}"
+  policy      = data.aws_iam_policy_document.rotate_iam_keys.json
+}
+
+resource "aws_iam_policy_role_attachment" "rotate_iam_keys" {
+  role       = var.rotation_lambda_role_arn
+  policy_arn = aws_iam_policy.rotate_iam_keys.arn
+}
+
+
 resource "aws_iam_access_key" "supplier" {
   user = aws_iam_user.supplier.name
 }
 
-resource "aws_secretsmanager_secret" "supplier" {
+module "secrets_manager" {
+  #checkov:skip=CKV_TF_1:Module registry does not support commit hashes for versions
+  source = "terraform-aws-modules/secrets-manager/aws"
+
   name        = "iam-${aws_iam_user.supplier.name}"
   description = "IAM user access credentials for ${var.data_feed}-${var.order_type}"
+  secret_string = jsonencode({
+    key    = aws_iam_access_key.supplier.id,
+    secret = aws_iam_access_key.supplier.secret
+  })
+  enable_rotation     = true
+  rotation_lambda_arn = var.rotation_lambda_arn
+  rotation_rules = {
+    automatically_after_days = 100
```

```suggestion
    automatically_after_days = 84
```

12 weeks, keeps it a consistent day of the week.